Do not install upstart job file
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,7 @@ clean-dpkg: clean
 
 copy-init:
 	install -m 755 dkms_autoinstaller debian/dkms.dkms_autoinstaller.init
-	install -m 755 dkms_autoinstaller.upstart debian/dkms.dkms_autoinstaller.upstart
+	#install -m 755 dkms_autoinstaller.upstart debian/dkms.dkms_autoinstaller.upstart
 
 install:
 	mkdir -m 0755 -p $(VAR) $(SBIN) $(MAN) $(INITD) $(ETC) $(BASHDIR) $(SHAREDIR) $(LIBDIR)
