--- a/dkms
+++ b/dkms
@@ -3005,7 +3005,9 @@
     ROOT=""
     ADMINABLE=""
     if [ "$USER" != "root" ]; then
-        if [ -x /usr/bin/gksudo ] && [ ! -z "$DISPLAY" ]; then
+        if [ -x /usr/sbin/su-to-root ] && [ ! -z "$DISPLAY" ]; then
+            ROOT="/usr/sbin/su-to-root -c"
+        elif [ -x /usr/bin/gksudo ] && [ ! -z "$DISPLAY" ]; then
             ROOT="/usr/bin/gksudo --description 'DKMS Debian package builder' "
         elif [ -x /usr/bin/kdesu ] && [ ! -z "$DISPLAY" ]; then
             ROOT="/usr/bin/kdesu"
