Description: Improve kernel postinst status information.
Author: Michael Gilbert <michael.s.gilbert@gmail.com>
--- a/kernel_postinst.d_dkms
+++ b/kernel_postinst.d_dkms
@@ -36,7 +36,12 @@ case "${uname_s}" in
 esac
 
 if [ -x /usr/lib/dkms/dkms_autoinstaller ]; then
-    exec /usr/lib/dkms/dkms_autoinstaller start $inst_kern > /dev/null
+    /usr/lib/dkms/dkms_autoinstaller start $inst_kern >&2
+fi
+
+if [ ! -e /lib/modules/$inst_kern/build/include ] ; then
+    echo "dkms: WARNING: linux headers are missing, which may explain the above failures." >&2
+    echo "      please install the linux-headers-$inst_kern package to fix this." >&2
 fi
 
 if ! _check_kernel_dir $inst_kern ; then
