--- a/dkms
+++ b/dkms
@@ -233,6 +233,9 @@ setup_kernels_arches()
 do_depmod()
 {
     # $1 = kernel version
+    if [ "${current_os}" != "Linux" ] ; then
+        return
+    fi
     if [[ -f /boot/System.map-$1 ]]; then
 	/sbin/depmod -a "$1" -F "/boot/System.map-$1"
     else
@@ -377,7 +380,7 @@ override_dest_module_location()
     local orig_location="$1"
     [[ ${addon_modules_dir} ]] && echo "/${addon_modules_dir}" && return
 
-    if [ "$uname_s" = "GNU/kFreeBSD" ] ; then
+    if [ "$current_os" = "GNU/kFreeBSD" ] ; then
 	# Does not support subdirs, regardless of distribution
 	echo "" && return
     fi
