Description: Upstream changes introduced in version 2.1.1.2-5
Author: Anders Kaseorg <andersk@mit.edu>
Bug-Debian: http://bugs.debian.org/755972
--- dkms-2.1.1.2.orig/dkms
+++ dkms-2.1.1.2/dkms
@@ -25,14 +25,14 @@ function invoke_command ()
     local exitval_file=`mktemp $tmp_location/dkms.XXXXXX`
     [ -z "$verbose" ] && echo -en "$2..." || echo -e "$1"
     if [ "$3" == background ] && [ -z "$verbose" ]; then
-        (eval $1 >/dev/null 2>&1; echo "exitval=$?" >> "$exitval_file") &
+        (eval "$1" >/dev/null 2>&1; echo "exitval=$?" >> "$exitval_file") &
         while [ -e "$exitval_file" ] && ! [ -s "$exitval_file" ]; do
             sleep 3
             echo -en "."
         done
         . "$exitval_file"
     else
-        eval $1; exitval=$?
+        eval "$1"; exitval=$?
     fi
     [ $exitval -gt 0 ] && echo -en "(bad exit status: $exitval)"
     rm -f "$exitval_file"
@@ -1524,7 +1524,7 @@ function do_build()
     echo $"`date`" >> "$dkms_tree/$module/$module_version/build/make.log"
     local the_make_command=`echo $make_command | sed "s/^make/make KERNELRELEASE=${kernelver_array[0]}/"`
 
-    invoke_command "$the_make_command >> $dkms_tree/$module/$module_version/build/make.log 2>&1" "$the_make_command" background
+    invoke_command "{$the_make_command;} >> $dkms_tree/$module/$module_version/build/make.log 2>&1" "$the_make_command" background
 
     # Make sure good return status
     if [ "$?" -ne 0 ]; then
